### PR TITLE
Provide more refined configuration of Devops agent

### PR DESCRIPTION
**Pull request by JohnNiang**
### What type of PR is this?
/area devops
/kind feature
### What this PR does / why we need it:
I provide three options for `devops.agent.config` configuration in ClusterConfiguration: default(or leave empty), high and custom, each representing a different limit level respectively, except for the custom option.

After doing this, 
- Users can configure different limit levels according to their own actual situation to improve the stability and efficiency of the system. 
- Once user has specified the custom option, any subsequent changes will not take effect.
- Even if uses upgrade from the old version, it will not have any impact.

**The main processing flow is as follow**:
devops agent config update procedure
 
### Which issue(s) this PR fixes:

None

### Special notes for reviewers:
```
Test docker image: johnniang/ks-installer:agent-config-level
```

### Does this PR introduced a user-facing change?

```release-note

```
### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: deploy/cluster-configuration.yaml [hunk 1]
```diff
@@ -58,6 +58,8 @@ spec:
     jenkinsJavaOpts_Xms: 512m  # The following three fields are JVM parameters.
     jenkinsJavaOpts_Xmx: 512m
     jenkinsJavaOpts_MaxRAM: 2g
+    agent:
+      config: default          # Available config level: default, high, custom
```
Comment (LinuxSuRen) [inline deploy/cluster-configuration.yaml:62]: Acorronding to the context, `default`, `hight`, and `custom` is not a level. It sounds like a name of a configuration.

IMO `config` could be anything. We'd better give an explicit one. And Jenkins isn't always the only CI engine in the future. For example:

```
devops:
  jenkinsAgent:
    podResourceLimit: default
```

File: roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+devops_agent:
+  concurrent: 20
```
Comment (LinuxSuRen) [inline roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml:2]: Why is 20 here?

`20 x 4096Mi = 81920Mi = 80Gi`

`3000m * 20 = 60cores`

Are you sure that's what we want? IMO, it is a very big request for many users who wants to adapt KubeSphere DevOps in their team.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Before this PR get merged, a document is necessary. Please create an issue to describe it.

File: controller/schedule.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@ else
   ansible-playbook /kubesphere/playbooks/telemetry.yaml -e @/kubesphere/config/ks-config.json
   if [[ $? -eq 0 ]]; then
     #statements
-    str="successsful!"
+    str="successful!"
```
[the remark this answers is not in the document]
Comment (JohnNiang) [reply controller/schedule.sh]: Thank you for pointing out, this change has noting to do with my feature. But I can confirm that the variable str is in code block scope, and the word `successsful` only appears once in current project. Finally, I agree with you, don't add irrelevant changes.

File: roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml (hunk 2 above)
Comment (JohnNiang) [reply roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml:2]: Thank you for your correction. I'll change them to appropriate value.

Comment (JohnNiang) [reply roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml:2]: But I also want to say what really plays a decisive role is the limit of PodTemplate and hardware resources.

File: deploy/cluster-configuration.yaml (hunk 1 above)
Comment (JohnNiang) [reply deploy/cluster-configuration.yaml:62]: The name looks like good to me.

File: roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml (hunk 2 above)
Comment (LinuxSuRen) [reply roles/ks-devops/jenkins/vars/jenkins_agent_limit_high.yaml]: > But I also want to say what really plays a decisive role is the limit of PodTemplate and hardware resources.

I didn't get your point here. Please explain it more.

**Comment by LinuxSuRen** [comment]
I believe that this PR can close now.

**Comment by JohnNiang** [comment]
Due to another implementation, we will close this issue later. Thanks for @LinuxSuRen 's mention.

/close

**Pull request closed without merging**